livestyled/models/product.py

```diff
@@ -86,12 +86,18 @@ def create_new(
 
     def diff(self, other):
         differences = {}
-        fields = (
-            'external_id', 'price', 'translations', 'product', 'tax', 'tax_rate', 'tax_band'
-        )
+        fields = ('external_id', 'price', 'translations', 'tax', 'tax_rate', 'tax_band')
         for field in fields:
             if getattr(self, field) != getattr(other, field):
                 differences[field] = getattr(self, field)
+
+        related_resource_fields = ['product']
+        for field in related_resource_fields:
```

Comment:
why is this this necessary? wouldn't python check the object structures and compare?

Reply:
Even if these objects body is equal - python will not say that objects are equal if one of them were created within our python sdk and the other one was created within our integration.

Such approach is used within our sdk as well, I saw it in several places.

Also, I think it's pretty logical that within this class we're checking differences of its body and its relations ids and not the relations object bodies. This class is responsible only about keeping the same relations when checking differences, but not its relations bodies